### DISCUSSION TITLE
Dudas varias sobre el enunciado

### DISCUSSION THREAD

**Discussion by RaniAgus** [dds-jv/foro#205] [TPA - 2021] [answered]
¡Buenas! Este fin de semana estuvimos viendo el enunciado del TPA y nos surgieron dudas que nos gustaría preguntarle a nuestro "cliente" (?)

Antes que nada, una duda sobre "nuestro sistema":
1. Dice que un `DueñoDeMascota` debe poder `Confirmar Mascota Encontrada`, pero ¿no puede hacer algo como `Notificar Mascota Perdida`? ¿Qué efecto tendría notificar a mi propia mascota como encontrada si en nuestro sistema no vamos a previamente marcarla como perdida?

Ahora sí, dudas sobre la primera iteración:

2. 
> Se debe permitir que un administrador agregue posibles características de mascotas con facilidad.

Asumimos que estas características las ingresa un administrador para que luego se puedan asociar a una mascota, así que estamos haciendo un `RepositorioCaracteristicas` para "almacenarlas". La pregunta es: ¿se debe poder `agregarCaracteristica` a una `Mascota` ya creada o todas las `Mascota`s que se creen cuentan con sus características para siempre, digamos?

3.
> Se debe permitir que una persona informe que encontró una mascota perdida con chapita para que un voluntario le notifique al dueño de esta situación.

Si un `Dueño` encuentra una `MascotaPerdida`, al entrar al formulario ¿aún así debe cargar todos sus datos aunque ya esté registrado como `Dueño`? ¿Está contemplada en un futuro la idea de considerar a un `Dueño` y a un `Rescatista` como una subcategoría de `Persona` o no tienen nada que ver entre sí?

4.
> Se le debe otorgar la posibilidad de generar un usuario a la persona que está registrando a su mascota en la plataforma.

Por "otorgar la posibilidad", ¿se refiere a que una `Mascota` puede tener un `Dueño` sin usuario, o a que al registrar tu `Mascota` podés elegir entre crear un `Usuario` y o loguearte con uno ya creado? ¿Pueden haber `Usuario`s que no sean `Dueño`s (por ejemplo, `Rescatista`s)?

¡Saludos!

**Comment by flbulgarelli** [comment, marked as answer]
¡Hola!

1. Por el momento la persona dueña de la mascota carga sus datos a fin de que si se pierde se la pueda contactar de alguna manera. En principio un dueño no va a poder informar que su mascota esta perdida.
2. Las mascotas ya creadas van a contar con las característica que existían en ese momento. Si quien administra agrega nuevas características las mismas van a tener efecto cuando se creen nuevas mascotas.
3. Sí, va a tener que cargar los datos nuevamente. A diferencia de las personas dueñas, quienes rescatan no son usuarios del sistema en el sentido de que tengan un registro y un login, ni un perfil, ni alguna otra idea que le de identidad persistente a quien encuentra a una mascota y luego otra. Por tanto probablemente no tenga sentido contar con una abstracción común entre ambas, al menos por ahora. 
4. Si no tenes cuenta en el sistema no vas a poder registrar una mascota, es condición para poder registrar una mascota. Lo de posibilidad no lo tomes literalmente porque es mas bien una obligación. 

¡Decinos si queda claro!

**Reply by RaniAgus** [reply]
¡Perfecto! Muchas gracias :)
